lms/djangoapps/discussion_api/render.py

```diff
@@ -0,0 +1,99 @@
+"""
+Content rendering functionality
+"""
+import re
+
+import markdown
+
+TAG_PATTERN = re.compile(r"<[^>]*>?")
+SANITIZED_TAG_PATTERN = re.compile(r"<(/?)(\w+)[^>]*>")
+ALLOWED_BASIC_TAG_PATTERN = re.compile(
+    r"^(</?(b|blockquote|code|del|dd|dl|dt|em|h1|h2|h3|i|kbd|li|ol|p|pre|s|sup|sub|strong|strike|ul)>|<(br|hr)\s?/?>)$"
+)
+ALLOWED_A_PATTERN = re.compile(
+    r'^(<a\shref="((https?|ftp)://|/)[-A-Za-z0-9+&@#/%?=~_|!:,.;\(\)]+"(\stitle="[^"<>]+")?\s?>|</a>)$'
+)
+ALLOWED_IMG_PATTERN = re.compile(
+    r'^(<img\ssrc="(https?://|/)[-A-Za-z0-9+&@#/%?=~_|!:,.;\(\)]+"(\swidth="\d{1,3}")?'
+    r'(\sheight="\d{1,3}")?(\salt="[^"<>]*")?(\stitle="[^"<>]*")?\s?/?>)$'
+)
```

Comment:
do these patterns need to allow for extraneous whitespace between the opening `<` (and `\`) and the start of the tag name?